### PR TITLE
ARROW-15976: [C++] Clean up commenting on execution plan example

### PR DESCRIPTION
**Pull request by westonpace**

### CODE REVIEW

**Comment by westonpace** [comment]
CC @vibhatha please take a look

**Comment by vibhatha** [comment]
Sure will do, @westonpace

**Comment by vibhatha** [comment]
Looks much better now and looks good to me. Sorry about my commenting style. Thanks a lot for working on this one.

**Review by vibhatha — APPROVED** [review]

File: cpp/examples/arrow/execution_plan_documentation_examples.cc [hunk 1]
```diff
@@ -579,14 +567,13 @@ arrow::Status SourceGroupAggregateSinkExample(cp::ExecContext& exec_context) {
...
- * This example shows how the data can be consumed within the execution plan
- * by using a ConsumingSink node. There is no data output from this execution plan.
- * \param exec_context : execution context
- * \return arrow::Status
- */
+
+/// \brief An example showing a consuming sink node
+/// \param exec_context The execution context to run the plan in
+///
+/// Source-ConsumingSink
```
Comment (edponce) [inline cpp/examples/arrow/execution_plan_documentation_examples.cc:574]: Nit: `Source-Consuming-Sink`

File: cpp/examples/arrow/execution_plan_documentation_examples.cc [hunk 2]
```diff
@@ -98,22 +98,21 @@ arrow::Result<std::shared_ptr<arrow::RecordBatch>> GetSampleRecordBatch(
...
-    5,null
-    6,false
-    7,false
-    8,true
- * \return arrow::Result<std::shared_ptr<arrow::Table>>
- */
+/// \brief Create a sample table
+/// The table's contents will be:
+/// a, b
+/// 1,null
```
Comment (edponce) [inline cpp/examples/arrow/execution_plan_documentation_examples.cc:104]: Nit: `a, b` has an extra whitespace.

File: cpp/examples/arrow/execution_plan_documentation_examples.cc [hunk 3]
```diff
@@ -641,16 +628,14 @@ arrow::Status SourceConsumingSinkExample(cp::ExecContext& exec_context) {
...
- * and the data is ordered in the sink node. The order can be
- * ASCENDING or DESCENDING and it is configurable. The output
- * is obtained as a table from the sink node.
- * \param exec_context : execution context
- * \return arrow::Status
- */
+/// \brief An example showing an order-by node
+/// \param exec_context The execution context to run the plan in
+///
+/// Source-OrderBySink
```
Comment (edponce) [inline cpp/examples/arrow/execution_plan_documentation_examples.cc:634]: Similar to previous: `Source-OrderBy-Sink`

**Review by edponce — APPROVED** [review]

**Comment by edponce** [comment]
Thank you for working on this. These are very helpful comments.

**Comment by vibhatha** [comment]
@edponce I will add a minor PR for these changes not sure if this was merged.

**Pull request closed without merging**